pkg/kv/kvserver/raft_log_queue.go

```diff
@@ -296,8 +296,8 @@ func newTruncateDecision(ctx context.Context, r *Replica) (truncateDecision, err
 		return truncateDecision{}, nil
 	}
 
-	// For all our followers, overwrite the RecentActive field (which is always
-	// true since we don't use CheckQuorum) with our own activity check.
+	// For all our followers, overwrite the RecentActive field with our own
+	// activity check.
```

Comment:
You could add to revisit this scheme when CheckQuorum is always on. Maybe not necessary.

Reply:
We can't trivially do so. Raft's `RecentActive` will be reset to `false` every couple of seconds (election timeout interval), and then flipped back to `true` on the next message from the follower. This periodic flipping back to false isn't going to work for our purposes.